gadget/install/install.go

```diff
@@ -202,46 +241,61 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options
 	if bootDevice == "" {
 		bootDevice, err = diskWithSystemSeed(laidOutBootVol)
 		if err != nil {
-			return nil, fmt.Errorf("cannot find device to create partitions on: %v", err)
+			return "", nil, nil, 0, fmt.Errorf("cannot find device to create partitions on: %v", err)
 		}
 	}
 
 	diskLayout, err := gadget.OnDiskVolumeFromDevice(bootDevice)
 	if err != nil {
-		return nil, fmt.Errorf("cannot read %v partitions: %v", bootDevice, err)
+		return "", nil, nil, 0, fmt.Errorf("cannot read %v partitions: %v", bootDevice, err)
 	}
 
 	// check if the current partition table is compatible with the gadget,
 	// ignoring partitions added by the installer (will be removed later)
 	if err := gadget.EnsureLayoutCompatibility(laidOutBootVol, diskLayout, nil); err != nil {
-		return nil, fmt.Errorf("gadget and system-boot device %v partition table not compatible: %v", bootDevice, err)
+		return "", nil, nil, 0, fmt.Errorf("gadget and system-boot device %v partition table not compatible: %v", bootDevice, err)
 	}
 
 	// remove partitions added during a previous install attempt
 	if err := removeCreatedPartitions(gadgetRoot, laidOutBootVol, diskLayout); err != nil {
-		return nil, fmt.Errorf("cannot remove partitions from previous install: %v", err)
+		return "", nil, nil, 0, fmt.Errorf("cannot remove partitions from previous install: %v", err)
 	}
 	// at this point we removed any existing partition, nuke any
 	// of the existing sealed key files placed outside of the
 	// encrypted partitions (LP: #1879338)
 	sealedKeyFiles, _ := filepath.Glob(filepath.Join(boot.InitramfsSeedEncryptionKeyDir, "*.sealed-key"))
 	for _, keyFile := range sealedKeyFiles {
 		if err := os.Remove(keyFile); err != nil && !os.IsNotExist(err) {
-			return nil, fmt.Errorf("cannot cleanup obsolete key file: %v", keyFile)
+			return "", nil, nil, 0, fmt.Errorf("cannot cleanup obsolete key file: %v", keyFile)
 		}
 	}
 
-	var created []gadget.OnDiskStructure
 	timings.Run(perfTimings, "create-partitions", "Create partitions", func(timings.Measurer) {
 		opts := &CreateOptions{
 			GadgetRootDir: gadgetRoot,
 		}
 		created, err = CreateMissingPartitions(diskLayout, laidOutBootVol, opts)
 	})
 	if err != nil {
-		return nil, fmt.Errorf("cannot create the partitions: %v", err)
+		return "", nil, nil, 0, fmt.Errorf("cannot create the partitions: %v", err)
 	}
 
+	bootVolGadgetName = laidOutBootVol.Name
+	bootVolSectorSize = diskLayout.SectorSize
+	return bootVolGadgetName, created, allLaidOutVols, bootVolSectorSize, nil
+}
+
+// Run creates partitions, encrypts them when expected, creates
+// filesystems, and finally writes content on them.
+func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options Options, observer gadget.ContentObserver, perfTimings timings.Measurer) (*InstalledSystemSideData, error) {
+	// Step 1: create partitions
+	bootVolGadgetName, created, allLaidOutVols, sectorSize, err :=
```

Comment:
I would rename sectorSize here too for clarity

Reply:
Done